libs/vkd3d/memory.c

```diff
@@ -678,8 +684,14 @@ static HRESULT vkd3d_select_memory_flags(struct d3d12_device *device, const D3D1
                     break;
                 case D3D12_CPU_PAGE_PROPERTY_WRITE_COMBINE:
                     *type_flags = VK_MEMORY_PROPERTY_HOST_VISIBLE_BIT | VK_MEMORY_PROPERTY_HOST_COHERENT_BIT;
+
+                    if (heap_properties->MemoryPoolPreference == D3D12_MEMORY_POOL_L1)
+                        *type_flags |= VK_MEMORY_PROPERTY_DEVICE_LOCAL_BIT;
+
                     if (vkd3d_config_flags & VKD3D_CONFIG_FLAG_FORCE_HOST_CACHED)
                         *type_flags |= VK_MEMORY_PROPERTY_HOST_CACHED_BIT;
+                    else if (heap_properties->MemoryPoolPreference == D3D12_MEMORY_POOL_L1)
```

Comment:
This looks kinda redundant to me?

Reply:
You're right, I think the top one snuck back in during one of the rebases.